batan-server/engine/game.ts

```diff
@@ -99,93 +194,494 @@ class Game {
+      this.players[player2Index].updatePlayerResources(false, randomResource, 1);
+    }
+  }
+
+  /**
+   * Draws a random development card from the bank and gives it to the player
+   * @param owner 
+   */
+  private drawRandomDevelopmentCard(owner: player){
+    let allDevCards: developmentType[] = [];
+    for(let i = 0; i < this.bank.developmentCards.knight; i++){
+      allDevCards.push(developmentType.knight)
+    } 
+    for(let i = 0; i < this.bank.developmentCards.monopoly; i++){
+      allDevCards.push(developmentType.monopoly)
+    } 
+    for(let i = 0; i < this.bank.developmentCards.roadBuilder; i++){
+      allDevCards.push(developmentType.roadBuilder)
+    } 
+    for(let i = 0; i < this.bank.developmentCards.victoryPointCard; i++){
+      allDevCards.push(developmentType.victoryPointCard)
+    } 
+    for(let i = 0; i < this.bank.developmentCards.yearOfPlenty; i++){
+      allDevCards.push(developmentType.yearOfPlenty)
+    } 
+    if(allDevCards.length > 0){
+      const randomIndex = Math.floor(Math.random() * allDevCards.length);
+      let randomDevCard = allDevCards[randomIndex];
+      this.giveDevCard(owner, randomDevCard);
+    }
+  }
+
+  private moveRobberAndSteal(user: player) {
+    //move robber
+      //TODO emit to player who rolled
+      let playerReply: string = this.getUserInput(`${player[user]}, where would you like to place the robber?\n`);
+      let hexId = parseInt(playerReply);
+      let affectedPlayers = this.board.moveRobber(hexId);
+
+      //steal resource
+      if (affectedPlayers.length > 0) {
+        let affectedPlayersString = "";
+        affectedPlayers.forEach(element => {
+          affectedPlayersString.concat(player[element] + " ");
+        })
+        //TODO emit to player who rolled
+        playerReply = this.getUserInput("Who would you like to steal from? " + affectedPlayersString + '\n');
+        let victim = player.none;
+        if (playerReply === "blue") {
+          victim = player.blue;
+        }
+        if (playerReply === "orange") {
+          victim = player.orange;
+        }
+        if (playerReply === "red") {
+          victim = player.red;
+        }
+        if (playerReply === "white") {
+          victim = player.white;
+        }
+        this.stealResource(user, victim);
+      }
+  }
+
+  /**
+   * Represents a player rolling the game dice. 
+   * @param user 
+   * @param roll 
+   */
+  rollDice(user: player/*, roll: number*/) { //TODO remove second param when not testing
+    const die1 = Math.floor(Math.random() * 6) + 1;
+    const die2 = Math.floor(Math.random() * 6) + 1;
     const diceRoll = die1 + die2;
+    //for testing: 
+    //const diceRoll = roll;
+
+    console.log(`${player[user]} rolled a ${diceRoll}!`);
 
     if (diceRoll === 7) {
-      //move robber logic and take resources
+      this.moveRobberAndSteal(user);
+
+      //discard if more than 7
+      this.players.forEach((player) => {
```

Comment:
Robber should be moved _after_ players discard half their cards if they have over 7

Reply:
Done, nice catch!